docs/resources/gke_cluster.md

```diff
@@ -0,0 +1,458 @@
+### Nested Schema for `spec.proxy`
+
+Optional:
+
+- `allow_insecure_bootstrap` (Boolean)
+- `bootstrap_ca` (String)
+- `enabled` (Boolean)
+- `http_proxy` (String)
+- `https_proxy` (String)
+- `no_proxy` (String)
+- `proxy_auth` (String)
+
+<a id="nestedblock--spec--sharing"></a>
+### Nested Schema for `spec.sharing`
+
+Optional:
+
+- `enabled` (Boolean)
+- `projects` (Block List) (see [below for nested schema](#nestedblock--spec--sharing--projects))
+
+<a id="nestedblock--spec--sharing--projects"></a>
+### Nested Schema for `spec.sharing.projects`
+
+Optional:
+
+- `name` (String)
+
+
+<a id="nestedblock--spec--config"></a>
+### Nested Schema for `spec.config`
+
+***Required***
+
+- `gcp_project` (String) GCP Project name.
+- `control_plane_version` (String) Kubernetes version of ControlPlane
+- `location` (Block List, Max: 1) GKE cluster location configuration. (see [below for nested schema](#nestedblock--spec--config--location))
+- `network` (Block List, Max: 1) GKE cluster Network configuration. (see [below for nested schema](#nestedblock--spec--config--network))
+- `node_pools` (Block List) GKE cluster node pool configuration. (see [below for nested schema](#nestedblock--spec--config--node_pools))
+
+Optional:
+
+- `features` (Block List, Max: 1) GKE cluster additional features configuration. (see [below for nested schema](#nestedblock--spec--config--features))
+- `pre_bootstrap_commands` (List of String) Commands will be executed every time Cluster nodes come up. Example: Node Creation, Node Restart.
+- `security` (Block List, Max: 1) GKE cluster Network configuration. (see [below for nested schema](#nestedblock--spec--config--security))
+
+
+<a id="nestedblock--spec--config--location"></a>
+### Nested Schema for `spec.config.location`
+
+***Required***
+
+- `type` (String) GKE Cluster location can be either `zonal` or `regional`
+- `config` (Block List, Max: 1) (see [below for nested schema](#nestedblock--spec--config--location--config))
+
+***Optional***
+
+- `default_node_locations` (Block List, Max: 1) For Zonal cluster increase availability by selecting more than one zone. For Regional cluster By default, Kubernetes Engine runs nodes of a regional cluster across three zones within a region. Select this option if you want to manually specify the zones in which this cluster's nodes run. All zones must be within the same region. (see [below for nested schema](#nestedblock--spec--config--location--default_node_locations))
+
+<a id="nestedblock--spec--config--location--config"></a>
+### Nested Schema for `spec.config.location.type`
+
+- `region` (String) Regional location in which the cluster's control plane and nodes are located. For ZonalCluster only zone information is sufficient. For Regional Cluster, both region and zone should be provided.
+- `zone` (String) Zone in the region where bootstrap VM is created for cluster provisioning. For ZonalCluster only zone information is sufficient. For Regional
```

Comment:
zone is used only for bootstrap vm?

Reply:
@kutumba-rafay Yes. When cluster type is `Regional`, we force user to provide a zone in that region and this zone is where we bring up bootstrap VM. Zone info is not used for anything else for regional cluster. 
Updated text. Hopefully it is clear now.